### ISSUE TITLE
Bug when running example in README.md

### ISSUE THREAD

**Issue by ThomasWollmann** [PyTorchLightning/pytorch-lightning#4367]
## 🐛 Bug

When running the example in the README.md, the script crashes (Error: AttributeError: 'TypeError' object has no attribute 'message') when calling "trainer.fit(...)".

Appeared with 1.0.3, 1.0.2, 1.0.1, 1.0.0. The script works in 0.9.0 if you remove the line "self.log('train_loss', loss)".

Stacktrace:
```
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:45: UserWarning: you passed in a val_dataloader but have no validation_step. Skipping validation loop
  warnings.warn(*args, **kwargs)
Traceback (most recent call last):
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 188, in log_metrics
    self.experiment.add_scalar(k, v, step)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/base.py", line 450, in experiment
    return get_experiment() or DummyExperiment()
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py", line 35, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/base.py", line 449, in get_experiment
    return fn(self)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 144, in experiment
    self._experiment = SummaryWriter(log_dir=self.log_dir, **self._kwargs)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 118, in log_dir
    version = self.version if isinstance(self.version, str) else f"version_{self.version}"
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 234, in version
    self._version = self._get_next_version()
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 246, in _get_next_version
    d = listing["name"]
TypeError: string indices must be integers
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/thomas/mx/tmp/main.py", line 41, in <module>
    trainer.fit(autoencoder, DataLoader(train), DataLoader(val))
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 440, in fit
    results = self.accelerator_backend.train()
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/accelerators/cpu_accelerator.py", line 45, in train
    self.trainer.train_loop.setup_training(model)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 132, in setup_training
    self.trainer.logger.log_hyperparams(ref_model.hparams)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py", line 35, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 169, in log_hyperparams
    self.log_metrics(metrics, 0)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py", line 35, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/thomas/anaconda3/lib/python3.7/site-packages/pytorch_lightning/loggers/tensorboard.py", line 191, in log_metrics
    type(e)(e.message + m)
AttributeError: 'TypeError' object has no attribute 'message'
```

### To Reproduce

```
import os
import torch
from torch import nn
import torch.nn.functional as F
from torchvision.datasets import MNIST
from torch.utils.data import DataLoader, random_split
from torchvision import transforms
import pytorch_lightning as pl

class LitAutoEncoder(pl.LightningModule):

    def __init__(self):
        super().__init__()
        self.encoder = nn.Sequential(nn.Linear(28 * 28, 128), nn.ReLU(), nn.Linear(128, 3))
        self.decoder = nn.Sequential(nn.Linear(3, 128), nn.ReLU(), nn.Linear(128, 28 * 28))
    
    def forward(self, x):
        # in lightning, forward defines the prediction/inference actions
        embedding = self.encoder(x)
        return embedding

    def training_step(self, batch, batch_idx):
        # training_step defined the train loop. It is independent of forward
        x, y = batch
        x = x.view(x.size(0), -1)
        z = self.encoder(x)
        x_hat = self.decoder(z)
        loss = F.mse_loss(x_hat, x)
        self.log('train_loss', loss)
        return loss

    def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
        return optimizer

dataset = MNIST(os.getcwd(), download=True, transform=transforms.ToTensor())
train, val = random_split(dataset, [55000, 5000])

autoencoder = LitAutoEncoder()
trainer = pl.Trainer()
trainer.fit(autoencoder, DataLoader(train), DataLoader(val))
```

### Expected behavior

Training should start.

### Environment

* CUDA:
	- GPU:
	- available:         False
	- version:           10.2
* Packages:
	- numpy:             1.17.2
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0
	- pytorch-lightning: 1.0.3
	- tqdm:              4.36.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
	- processor:         x86_64
	- python:            3.7.4
	- version:           #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020

**Comment by rohitgr7** [comment]
Tried the code on colab. Not getting any error.

**Comment by Maddy12** [comment]
I am getting the same error

**Comment by rohitgr7** [comment]
Can you guys reproduce this error with a colab notebook??

**Comment by bipinKrishnan** [comment]
> Tried the code on colab. Not getting any error.

Same here, the code works fine on Google Colab.

**Comment by ThomasWollmann** [comment]
Fine for me in Colab. Still not working locally.

**Comment by 21jun** [comment]
I solved this problem by installing pytorch-lightning v.1.0.5
```sh
pip uninstall pytorch-lightning
pip install pytorch-lighting=1.0.5
```


In Google Colab, pytorch-lightning version is 1.0.5 while my local version was 0.9
At the first place, I installed pytorch-lightning with `conda install pytorch-lightning -c conda-forge`  but it gave me v 0.9
```
OS: Windows 10
arch: x86_64
conda: 4.8.5
python: 3.8.2
pytorch: 1.6.0
```

**Comment by ThomasWollmann** [comment]
I did the same. Also installed pytorch via conda. Still the same error.

**Comment by edenlightning** [comment]
@Borda any idea?

**Comment by Borda** [comment]
@awaelchli can we add running doctest (via Sphinx) also on all MarkDown files, what do you think? :]

**Comment by awaelchli** [comment]
that's probably a bit overkill :)

**Comment by ThomasWollmann** [comment]
@Borda instead having the samples as regular unit tests would be already nice. Do not thing there is not an urgent need for doctest.

**Comment by Borda** [comment]
> @Borda having the samples as regular unit tests would be already nice. Do not thing there is not an urgent need for doctest.

well, how would utilize unit-tests as samples? the point with doctsring is that are examples and also tested so one code serves twice, compare to unit-tests which are usually in another folder (out of the package distribution) and then you just hope that if you a change in tests you update example accordingly...

**Comment by ThomasWollmann** [comment]
> > @Borda having the samples as regular unit tests would be already nice. Do not thing there is not an urgent need for doctest.
> 
> well, how would utilize unit-tests as samples? the point with doctsring is that are examples and also tested so one code serves twice, compare to unit-tests which are usually in another folder (out of the package distribution) and then you just hope that if you a change in tests you update example accordingly...

That's true.

**Comment by ThomasWollmann** [comment]
I reinstalled my OS and couldn't reproduce the error anymore. Will close the issue for now.

**Comment by Jerryxiaoyu** [comment]
@edenlightning  I got the same problem. I fixed it by reinstalling a higher version TensorFlow (1.14+). 
I think this is because Pytorch-lighting requires TensorBoard >=2.2， but that means also requires a higher version TensorFlow to support the TensorBoard. 
I found the TensorFlow version was only 1.12, so I upgraded it, and no errors.

https://github.com/tensorflow/tensorboard#can-i-run-tensorboard-without-a-tensorflow-installation
